staging/src/k8s.io/apiserver/pkg/server/filters/timeout.go

```diff
@@ -97,6 +106,7 @@ func (t *timeoutHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	case <-done:
 		return
 	case <-after:
+		cancelFunc()
```

Comment:
Is there a technical difference between having people just include their `cancelFunc` logic inside of the `recordFn`?  Maybe change it's name, but appears logically equivalent from a code perspective.

Reply:
No, there isn't. The name `recordFn` is misleading. We should call it `onTimeoutFn` and cover both, metrics and cancellation.

Reply:
Fixed.